### PR TITLE
Add support for clang tidy 19

### PR DESCRIPTION
**Pull request by ChillerDragon**

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -31,6 +31,10 @@ Checks: >
   -bugprone-suspicious-include,
   -bugprone-unhandled-self-assignment,
   -bugprone-suspicious-realloc-usage,
+  -bugprone-switch-missing-default-case,
+  -bugprone-casting-through-void,
+  -bugprone-multi-level-implicit-pointer-conversion,
+  -bugprone-unchecked-optional-access,
```
Comment (Robyt3) [inline .clang-tidy:37]: Does this happen frequently in our code? I don't think it should be excluded either or it should have a comment explaining that it's a temporary exclusion.

File: .clang-tidy [hunk 2]
```diff
@@ -85,8 +90,13 @@ Checks: >
   -readability-simplify-boolean-expr,
   -readability-suspicious-call-argument,
   -readability-uppercase-literal-suffix,
+  -readability-use-std-min-max,
+  -readability-redundant-casting,
```
Comment (Robyt3) [inline .clang-tidy:94]: What's the reason for excluding this check? Seems like a safe check to improve readability.

File: .clang-tidy (hunk 1 above)
Comment (ChillerDragon) [reply .clang-tidy:37]: I think thats only m_Score but it happend like a few times so I did not feel like spamming in a bunch of if statements. Seems like an unwanted change.

File: .clang-tidy (hunk 2 above)
Comment (ChillerDragon) [reply .clang-tidy:94]: https://github.com/ddnet/ddnet/blob/d4e3f6fc01b594e18c30b85f256deac4760be45b/src/game/collision.cpp#L1158

I did not feel comfortable removing that one.

Comment (ChillerDragon) [reply .clang-tidy:94]: But yea that is probably okay to be removed. If you say its good ill drop the cast.

File: .clang-tidy [hunk 3]
```diff
@@ -85,8 +90,13 @@ Checks: >
...
+  -readability-static-accessed-through-instance,
```
[the remark this answers is not in the document]
Comment (ChillerDragon) [reply .clang-tidy:95]: Not needed anymore since https://github.com/ddnet/ddnet/pull/8317 is merged. I will drop it.

**Comment by ChillerDragon** [comment]
Okay more redundant casts came up. I will have to fix that at some point.

File: .clang-tidy (hunk 2 above)
Comment (Robyt3) [reply .clang-tidy]: Seems fine, `i` is already an `int` and `d` is a `float`.

File: .clang-tidy (hunk 1 above)
Comment (heinrich5991) [reply .clang-tidy:37]: Can you give an example of that warning?

Comment (ChillerDragon) [reply .clang-tidy:37]:
```
/home/chiller/Desktop/git/ddnet/src/game/server/player.cpp:940:19: error: unchecked access to optional value [bugprone-unchecked-optional-access,-warnings-as-errors]
  940 |                         str_time_float(Result.m_Data.m_Info.m_Time.value(), TIME_HOURS_CENTISECS, aTime, sizeof(aTime));
      |                                        ^
```

```
/home/chiller/Desktop/git/ddnet/src/engine/server/server.cpp:2092:13: error: unchecked access to optional value [bugprone-unchecked-optional-access,-warnings-as-errors]
 2092 |                                 Score = m_aClients[i].m_Score.value();
      |                                         ^
```

Comment (heinrich5991) [reply .clang-tidy:37]: Yea, especially the latter one looks like a false positive. Sucks. Thanks for the examples.

**Review by Robyt3 — CHANGES_REQUESTED** [review]
Looks good, but some changes got mixed up in the commits.

The `if(sock->ipv4sock < 0)` and `if(sock->ipv6sock < 0)` checks should be added in a separate commit, I don't see how they are related to `modernize-use-emplace`.

The removal of explicit `std::make_pair` probably belongs in the commit that fixes `modernize-use-emplace`, instead of being included in 8025cc01f7b1b9828a0d054864844ab7610d27ba.

**Review by Robyt3 — APPROVED** [review]

**Pull request merged**